### PR TITLE
Pass system-probe test packages through read-only directory

### PR DESCRIPTION
**Pull request by usamasaqib**
### What does this PR do?
This PR shares the system-probe test packages through a read-only filesystem mounted in each KMT micro-vm.

### Motivation
NPM reported an issue with the `TestConntrackCompile` test. The test has a high degree of flakiness in KMT.
image

The problem happened due to high disk usage during KMT setup, which led to high IO wait times stemming from the large number of requests the underlying host had to service. From observing IO statistics we were able to correlate the problematic operations to be the copying of system-probe test packages to the VM's filesystem.

This is illustrated in the graph below. It graphs transactions per second (tps), % cpu spent waiting on IO (iowait) and kilobytes written per second (kB_wrtn/s). We have sharp spikes in all three metrics between `450` seconds - `600` seconds.
The thick black vertical lines represent the start time for the tests which had `TestConntrackCompile` failures. As can be seen the failures happen when the tests start times align with the spike in IO activity, which corresponds to copying the test packages.

image

When we eliminate the copy of system-probe tests to the VMs by sharing these via a read-only mounted filesystem with each VM, IO activity is significantly reduced following the initial environment setup. As we can see in the graph below, the previous spike in the various metrics disappears.
image

### Additional Notes
https://datadoghq.atlassian.net/browse/NPM-3107
https://datadoghq.atlassian.net/browse/EBPF-355

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided. Except if the `qa/skip-qa` label, with required either `qa/done` or `qa/no-code-change` labels, are applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: .gitlab/kernel_version_testing/system_probe.yml [hunk 1]
```diff
@@ -84,34 +84,22 @@ pull_test_dockers_arm64:
...
-    - pushd $DD_AGENT_TESTING_DIR/$ARCH
-    - tar czvf ../$ARCHIVE_NAME dependencies
-    - popd
-    # upload tests
+    # upload dependencies
     - !reference [.wait_for_instances]
     - !reference [.get_instance_ip_by_type]
     - !reference [.setup_ssh_config]
-    - scp $DD_AGENT_TESTING_DIR/$ARCHIVE_NAME metal_instance:/opt/kernel-version-testing/
+    - chmod +x $CI_PROJECT_DIR/test/new-e2e/system-probe/test/micro-vm-init.sh
```
Comment (brycekahle) [inline .gitlab/kernel_version_testing/system_probe.yml:91]: You should be able to fix the mode of the file in the repo and not need this?

File: pkg/network/protocols/tls/java/jattach_test.go [hunk 2]
```diff
@@ -19,8 +19,19 @@ import (
 	javatestutil "github.com/DataDog/datadog-agent/pkg/network/protocols/tls/java/testutil"
 	"github.com/DataDog/datadog-agent/pkg/network/testutil"
 	"github.com/DataDog/datadog-agent/pkg/util/kernel"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/cihub/seelog"
```
Comment (guyarb) [inline pkg/network/protocols/tls/java/jattach_test.go:23]: nit: move `"github.com/cihub/seelog"` to a different section

**Review by guyarb — APPROVED** [review]
Approve with a single nit comment

File: pkg/network/protocols/tls/java/jattach_test.go [hunk 3]
```diff
@@ -14,13 +14,24 @@ import (
...
 	"github.com/stretchr/testify/require"
 
 	javatestutil "github.com/DataDog/datadog-agent/pkg/network/protocols/tls/java/testutil"
 	"github.com/DataDog/datadog-agent/pkg/network/testutil"
 	"github.com/DataDog/datadog-agent/pkg/util/kernel"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
 )
 
+func TestMain(m *testing.M) {
```
Comment (brycekahle) [inline pkg/network/protocols/tls/java/jattach_test.go:26]: added for debug?

Comment (usamasaqib) [reply pkg/network/protocols/tls/java/jattach_test.go:26]: Yes. But it is useful to keep here IMO, so we can easily enable debug logs for java tests.

File: pkg/network/protocols/tls/java/jattach_test.go [hunk 4]
```diff
@@ -98,7 +109,7 @@ func TestInject(t *testing.T) {
 func TestInjectInReadOnlyFS(t *testing.T) {
 	curDir, err := os.Getwd()
 	require.NoError(t, err)
-	rodir := filepath.Join(curDir, "rodir")
+	rodir := filepath.Join("/", "tmp", "rodir")
```
[the remark this answers is not in the document]
Comment (usamasaqib) [reply pkg/network/protocols/tls/java/jattach_test.go]: The cleanup functions fail when using `t.TempDir`, on machines where the tests are skipped. `t.TempDir` has some side effect due to which the unmount fails in this case only.
https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/410651193

Comment (brycekahle) [inline pkg/network/protocols/tls/java/jattach_test.go:112]: nit: use `os.TempDir()` instead of assuming `/tmp`.

**Review by brycekahle — APPROVED** [review]

**Pull request merged**